tests/languages/docker_test.py

```diff
@@ -62,6 +62,25 @@ def invalid_attribute():
         assert docker.get_docker_user() == ()
 
 
+@pytest.mark.parametrize(
+    ('info_ret', 'expect_root'),
+    (
+        ((0, b'["name=rootless","name=cgroupns"]', b''), True),
+        ((0, b'["name=cgroupns"]', b''), False),
+        ((1, b'', b''), False),
+    ),
+)
+def test_docker_user_rootless_docker(info_ret, expect_root):
+    docker._is_rootless_docker.cache_clear()
+    with mock.patch.object(
+            docker, 'cmd_output_b', return_value=info_ret,
+    ):
+        if expect_root:
+            assert docker.get_docker_user() == ()
+        else:
+            assert docker.get_docker_user() != ()
```

Comment:
don't write logic in tests -- these are two separate disparate behaviours and should be tested separately

Reply:
> don't write logic in tests -- these are two separate disparate behaviours and should be tested separately

[909c165](https://github.com/pre-commit/pre-commit/pull/3446/commits/909c165194672b5e998c6bbd608277f14f26b08e) also included your cache suggestion from above